test/generator/test_generator.py

```diff
@@ -88,3 +92,90 @@ def test_distribute_remaining_items_no_locations_left(
         InvalidConfiguration, match=r"Received 881 remaining pickups, but there's only \d+ unassigned locations."
     ):
         generator._distribute_remaining_items(rng, filler_results, [default_echoes_preset])
+
+
+@pytest.mark.parametrize(
+    "logical_placement_cases",
+    [
+        (
+            LogicalPickupPlacementConfiguration.MINIMAL,
+            "Trivial",
+        ),
+        (
+            LogicalPickupPlacementConfiguration.MAJORS,
+            "Major ≥ 1",
+        ),
+        (
+            LogicalPickupPlacementConfiguration.ALL,
+            "(Major ≥ 1 and Minor ≥ 1)",
+        ),
+    ],
+)
+def test_victory_condition_for_pickup_placement(
+    mocker,
+    blank_game_description,
+    blank_pickup_database,
+    default_generator_params,
+    default_generator_params_minor,
+    logical_placement_cases,
+):
+    # Setup
+    mocker.patch.object(blank_game_description, "victory_condition", Requirement.trivial())
+
+    pickups = [
+        PickupEntry(
+            name="Major Item",
+            model=PickupModel(
+                game=blank_game_description.game,
+                name="EnergyTransferModule",
+            ),
+            pickup_category=blank_pickup_database.pickup_categories["gear"],
+            broad_category=blank_pickup_database.pickup_categories["gear"],
+            progression=(
+                (
+                    ItemResourceInfo(
+                        resource_index=2,
+                        long_name="Major",
+                        short_name="major",
+                        max_capacity=1,
+                    ),
+                    1,
+                ),
+            ),
+            generator_params=default_generator_params,
+            resource_lock=None,
+            unlocks_resource=False,
+        ),
+        PickupEntry(
+            name="Minor Item",
+            model=PickupModel(
+                game=blank_game_description.game,
+                name="EnergyTransferModule",
+            ),
+            pickup_category=blank_pickup_database.pickup_categories["ammo"],
+            broad_category=blank_pickup_database.pickup_categories["ammo"],
+            progression=(
+                (
+                    ItemResourceInfo(
+                        resource_index=3,
+                        long_name="Minor",
+                        short_name="minor",
+                        max_capacity=1,
+                    ),
+                    1,
+                ),
+            ),
+            generator_params=default_generator_params_minor,
+            resource_lock=None,
+            unlocks_resource=False,
+        ),
+    ]
+
+    # Run
+    victory_str = str(
+        generator.victory_condition_for_pickup_placement(pickups, blank_game_description, logical_placement_cases[0])
```

Comment:
rather than indexing `logical_placement_cases` like this, put this line at the top of the test function: `placement_config, expected = logical_placement_cases` and then use the named variables here and in the assert